jolt-core/src/subprotocols/hamming_weight.rs

```diff
@@ -0,0 +1,284 @@
+use allocative::Allocative;
+#[cfg(feature = "allocative")]
+use allocative::FlameGraphBuilder;
+use ark_std::Zero;
+use rayon::prelude::*;
+use std::ops::Deref;
+use std::ops::DerefMut;
+use std::{cell::RefCell, rc::Rc};
+
+use crate::{
+    field::{JoltField, MaybeAllocative, MulTrunc},
+    poly::{
+        multilinear_polynomial::{BindingOrder, MultilinearPolynomial, PolynomialBinding},
+        opening_proof::{
+            OpeningAccumulator, OpeningPoint, ProverOpeningAccumulator, SumcheckId,
+            VerifierOpeningAccumulator, BIG_ENDIAN,
+        },
+    },
+    transcripts::Transcript,
+    zkvm::witness::CommittedPolynomial,
+};
+
+use crate::subprotocols::sumcheck::SumcheckInstance;
+
+#[derive(Allocative)]
+pub struct HammingWeightProverState<F: JoltField> {
+    /// ra polynomials
+    pub ra: Vec<MultilinearPolynomial<F>>,
+}
+
+/// Configuration trait for hamming weight sumchecks
+pub trait HammingWeightConfig {
```

Comment:
Why do we need separate config trait? Why don't we just put these methods on the `HammingWeightSumcheck` trait?